### DISCUSSION TITLE
Easy way to search all files

### DISCUSSION THREAD

**Discussion by drk-mtr** [BurntSushi/ripgrep#1669] [General] [answered]
Thanks for the brilliant software, really useful - and fast!

I would love it if there was an easy way to simply include all files.

I frequently use ripgrep to try and search a folder, but I rarely have full faith in its result, because I don't know whether I have successfully included all files. It is usually being too clever for me, by excluding files in the `.gitignore`, excluding files that start with `.` etc. etc.

I'm sure there probably is a simply way to include all files... but it's not immediately obvious to me.

Apologies in advance if this is already an option and I've simply missed it... I'm pretty sure it'll turn out that this is the case - is there a single switch that will include all files?

**Comment by BurntSushi** [comment]
It's explained here: https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering

It should also be in the man page and the `--help` output.

**Comment by dmlary** [comment, marked as answer]
I'm expanding on this answer for anyone else landing here from a search.

**The flag you need is `-uuu`**.

I'm adding this because the link in the accepted answer is two paragraphs explaining the smart searching in depth before describing the "--unrestricted" flag.  Also, for finding it in the `--help` info, I searched for phrases like "ignore" (as in ignore .gitignore), or "all" (as in search all files), not the description used for the `--unrestricted` flag, which is "Reduce the level of "smart" searching."
